File: CS/DZ 14.09.22 (Static, Polymorphism)/CS_Polymorphism/CS_Polymorphism/Rectangle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CS_Polymorphism
{
    class Rectangle: Figure
    {
        protected int a, b;

        public Rectangle(int x, int y, int a, int b): base(x, y)
        {
            this.a = a;
            this.b = b;
        }

        public override double GetS()
        {
            return a * b;
        }

        public override void Print()
        {
            // вызов метода Print базового класса
            //base.Print();

            Console.WriteLine($"Rectangle. x = {x}, y = {y}, a = {a}, b = {b}");
        }
    }
}

File: C++/ДЗ 10.06.22(полиморфизм_2)/CPP_Inheritance/CPP_Inheritance/Rectangle.cpp
#include "Rectangle.h"

#include <iostream>

using namespace std;

Rectangle::Rectangle(int x, int y, int a, int b): Figure(x, y)
{
	cout << "Rectangle constr" << endl;
	A = a;
	B = b;
}

Rectangle::~Rectangle()
{
	cout << "Rectangle destr" << endl;
}

void Rectangle::Print()
{
	//Figure::Print();
	cout << "Rectangle. X = " << X << ", Y = " << Y << ", A = " << A << ", B = " << B << endl;
}